src/main/java/util/events/UpdateRemainingRateEvent.java

```diff
@@ -0,0 +1,12 @@
+package util.events;
+
+/**
+ * An event to call tasks related to the api remaining rate.
+ */
+public class UpdateRemainingRateEvent extends UpdateRateLimitsEvent{
```

Comment:
An event is something that happens, not something that needs to be done. 'timer expired' is an event. 'refresh completed' is an event. 'update the rate' is an action.